packages/dappmanager/src/modules/chains/drivers/ethereum2.ts

```diff
@@ -0,0 +1,145 @@
+import fetch from "node-fetch";
+import memoize from "memoizee";
+import { getPrivateNetworkAlias } from "../../../domains";
+import { urlJoin } from "../../../utils/url";
+import { InstalledPackageData } from "../../../types";
+import { ChainDataResult } from "../types";
+import { safeProgress } from "../utils";
+
+const beaconChainServiceName = "beacon-chain";
+const MIN_SLOT_DIFF_SYNC = 60;
+
+// Wait for Promises to resolve. Do not cache rejections
+// Cache for 1 hour, genesis and config should never change
+const fetchNodeSyncingMemo = memoize(fetchNodeSyncingStatus, {
+  promise: true,
+  maxAge: 3e6
+});
+
+/**
+ * Returns a chain data object for an Ethereum 2.0 Prysm beacon chain node
+ * @param apiUrl = "http://beacon-chain.prysm-pyrmont.dappnode:3500/"
+ */
+export async function ethereum2(
+  dnp: InstalledPackageData
+): Promise<ChainDataResult | null> {
+  const beaconChainContainer = dnp.containers.find(
+    container => container.serviceName === beaconChainServiceName
+  );
+  if (!beaconChainContainer) {
+    throw Error(`${beaconChainServiceName} service not found`);
+  }
+  if (!beaconChainContainer.running) {
+    return null; // OK to not be running, just ignore
+  }
+
+  const containerDomain = getPrivateNetworkAlias(beaconChainContainer);
+
+  // http://beacon-chain.prysm-pyrmont.dappnode:3500/
+  const apiUrl = `http://${containerDomain}:3500`;
```

Comment:
Warning! Each client has it's own port. 3500 only applies to Prysm. Each client should declare the port in the manifest

Reply:
implement on https://github.com/dappnode/DNP_DAPPMANAGER/pull/963